### PR TITLE
virsh_domstate: virsh resume required for guest to reboot preserved vm

### PR DESCRIPTION
**Pull request by balamuruhans**
with on_crash as preserve, guest will be in crashed (panicked) state after
triggering crash, with virsh reset guest state changes to paused (crashed)
and requires virsh resume for reboot to proceed.

https://bugzilla.redhat.com/show_bug.cgi?id=1269575

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
**Job log**
```
JOB ID     : 7a4f8c59d7f0ce6c4a0290b1571c702ea6b3d510
JOB LOG    : /home/bala/avocado-fvt-wrapper/results/job-2018-02-19T16.11-7a4f8c5/job.log
 (1/1) guest_lifecycle.domstate.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.domstate.normal_test.reason.crash_vm.oncrash_preserve.reset: PASS (75.37 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 79.33 s
JOB HTML   : /home/bala/avocado-fvt-wrapper/results/job-2018-02-19T16.11-7a4f8c5/results.html

```

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py [hunk 1]
```diff
@@ -279,9 +280,15 @@ def run(test, params, env):
                     if vm_oncrash_action == "preserve" and reset_action:
                         virsh_dargs = {'debug': True, 'ignore_status': True}
                         ret = virsh.reset(vm_name, **virsh_dargs)
-                        if ret.exit_status != 0:
-                            test.fail("virsh reset failed after guest crash: "
-                                      "%s" % ret.stderr)
+                        libvirt.check_exit_status(ret)
+                        if "paused" not in virsh.domstate(vm_name, extra,
+                                                          **virsh_dargs).stdout.strip():
+                            test.fail("vm fails to change state from crashed"
```
Comment (sathnaga) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py]: it is better to explicitly check `paused (crashed)`

Comment (balamuruhans) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_domstate.py]: Agreed, I will make the change suggested.

**Comment by balamuruhans** [comment]
@sathnaga made the changes as suggested.

**Comment by sathnaga** [comment]
@balamuruhans pls post the test result,

**Comment by balamuruhans** [comment]
**Job Log**
```
JOB ID     : faab3d2f3388468e79034654e21ea67a743d6294
JOB LOG    : /home/bala/avocado-fvt-wrapper/results/job-2018-02-20T17.52-faab3d2/job.log
 (1/1) guest_lifecycle.domstate.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.domstate.normal_test.reason.crash_vm.oncrash_preserve.reset:  PASS (83.90 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 88.50 s
JOB HTML   : /home/bala/avocado-fvt-wrapper/results/job-2018-02-20T17.52-faab3d2/results.html

```
**Log:**
```
2018-02-20 17:52:57,887 client           L1078 DEBUG| Sending command: service kdump stop
2018-02-20 17:52:58,133 client           L1078 DEBUG| Sending command: echo $?
2018-02-20 17:52:58,234 client           L1078 DEBUG| Sending command: echo 1 > /proc/sys/kernel/sysrq
2018-02-20 17:52:58,336 client           L1078 DEBUG| Sending command: echo $?
2018-02-20 17:52:58,437 client           L1078 DEBUG| Sending command: echo c > /proc/sysrq-trigger
2018-02-20 17:53:01,600 virsh            L0648 DEBUG| Running virsh command: domstate virt-tests-vm1-lifecycle --reason
2018-02-20 17:53:01,600 process          L0523 INFO | Running '/bin/virsh domstate virt-tests-vm1-lifecycle --reason'
2018-02-20 17:53:01,630 process          L0367 DEBUG| [stdout] crashed (panicked)
2018-02-20 17:53:01,631 process          L0605 INFO | Command '/bin/virsh domstate virt-tests-vm1-lifecycle --reason' finished with 0 after 0.0259981155396s
2018-02-20 17:53:01,631 process          L0367 DEBUG| [stdout]
2018-02-20 17:53:01,648 virsh            L0696 DEBUG| status: 0
2018-02-20 17:53:01,649 virsh            L0697 DEBUG| stdout: crashed (panicked)
2018-02-20 17:53:01,649 virsh            L0698 DEBUG| stderr:
2018-02-20 17:53:01,649 virsh_domstate   L0040 INFO | Expected state: crashed (panicked)
2018-02-20 17:53:01,649 virsh_domstate   L0053 DEBUG| Actual state: crashed (panicked)
2018-02-20 17:53:01,649 virsh            L0648 DEBUG| Running virsh command: reset virt-tests-vm1-lifecycle
2018-02-20 17:53:01,650 process          L0523 INFO | Running '/bin/virsh reset virt-tests-vm1-lifecycle'
2018-02-20 17:53:01,675 process          L0367 DEBUG| [stdout] Domain virt-tests-vm1-lifecycle was reset
2018-02-20 17:53:01,675 process          L0605 INFO | Command '/bin/virsh reset virt-tests-vm1-lifecycle' finished with 0 after 0.0209898948669s
2018-02-20 17:53:01,675 process          L0367 DEBUG| [stdout]
2018-02-20 17:53:01,698 virsh            L0696 DEBUG| status: 0
2018-02-20 17:53:01,699 virsh            L0697 DEBUG| stdout: Domain virt-tests-vm1-lifecycle was reset
2018-02-20 17:53:01,699 virsh            L0698 DEBUG| stderr:
2018-02-20 17:53:01,699 libvirt          L1560 DEBUG| Command output:
Domain virt-tests-vm1-lifecycle was reset
2018-02-20 17:53:01,699 virsh            L0648 DEBUG| Running virsh command: domstate virt-tests-vm1-lifecycle --reason
2018-02-20 17:53:01,700 process          L0523 INFO | Running '/bin/virsh domstate virt-tests-vm1-lifecycle --reason'
2018-02-20 17:53:01,726 process          L0367 DEBUG| [stdout] paused (crashed)
2018-02-20 17:53:01,726 process          L0605 INFO | Command '/bin/virsh domstate virt-tests-vm1-lifecycle --reason' finished with 0 after 0.0221409797668s
2018-02-20 17:53:01,727 process          L0367 DEBUG| [stdout]
2018-02-20 17:53:01,748 virsh            L0696 DEBUG| status: 0
2018-02-20 17:53:01,749 virsh            L0697 DEBUG| stdout: paused (crashed)
2018-02-20 17:53:01,749 virsh            L0698 DEBUG| stderr:
2018-02-20 17:53:01,749 virsh            L0648 DEBUG| Running virsh command: resume virt-tests-vm1-lifecycle
2018-02-20 17:53:01,750 process          L0523 INFO | Running '/bin/virsh resume virt-tests-vm1-lifecycle'
2018-02-20 17:53:01,781 process          L0367 DEBUG| [stdout] Domain virt-tests-vm1-lifecycle resumed
2018-02-20 17:53:01,781 process          L0605 INFO | Command '/bin/virsh resume virt-tests-vm1-lifecycle' finished with 0 after 0.0273611545563s
2018-02-20 17:53:01,782 process          L0367 DEBUG| [stdout]
2018-02-20 17:53:01,807 virsh            L0696 DEBUG| status: 0
2018-02-20 17:53:01,807 virsh            L0697 DEBUG| stdout: Domain virt-tests-vm1-lifecycle resumed
2018-02-20 17:53:01,807 virsh            L0698 DEBUG| stderr:
2018-02-20 17:53:01,808 libvirt          L1560 DEBUG| Command output:
Domain virt-tests-vm1-lifecycle resumed

```

**Pull request merged**